### PR TITLE
Fix some named mutex tests

### PR DESCRIPTION
**Pull request by kouvel**
This reverts commit 697bd95eb1182b73225d4aede89647ae0f65f1f3.

Equivalent to PR #8625

### CODE REVIEW

File: src/System.Threading/tests/MutexTests.cs [hunk 1]
```diff
@@ -130,34 +126,47 @@ public void OpenExisting_NameUsedByOtherSynchronizationPrimitive_Windows()
...
+                switch (waitType)
+                {
+                    case 0: // WaitOne
+                        Assert.Throws<AbandonedMutexException>(() => m.WaitOne(FailedWaitTimeout));
+                        break;
+
+                    case 1: // WaitAny
+                        AbandonedMutexException ame = Assert.Throws<AbandonedMutexException>(() => WaitHandle.WaitAny(new[] { m }, FailedWaitTimeout));
+                        Assert.Equal(0, ame.MutexIndex);
+                        Assert.Equal(m, ame.Mutex);
```
Comment (stephentoub) [inline src/System.Threading/tests/MutexTests.cs]: Do these asserts apply to case 0 as well?  If so, they could be moved to after the switch block.

**Comment by stephentoub** [comment]
One comment, otherwise LGTM.  Presumably this won't pass CI until packages are updated?

File: src/System.Threading/tests/MutexTests.cs (hunk 1 above)
Comment (kouvel) [inline src/System.Threading/tests/MutexTests.cs]: It doesn't look like it, we only throw an exception with the extra mutex info for WaitAny.

**Comment by kouvel** [comment]
Yea is there something I should do to update packages, or are they updated automatically periodically?

**Comment by stephentoub** [comment]
> Yea is there something I should do to update packages, or are they updated automatically periodically?

I think @ericstj and @ellismg are working on it.

File: src/System.Threading/tests/MutexTests.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Threading/tests/MutexTests.cs]: Ok. Thanks.

**Comment by ellismg** [comment]
> I think @ericstj and @ellismg are working on it.

Correct, going to take another half a day or so.

**Comment by kouvel** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Pull request merged**